benchmarks/profiler/utils/config_modifiers/parallelization_mapping.py

```diff
@@ -0,0 +1,170 @@
+    verified against model properties.
+
+    Verification rules:
+    - TP and TEP must divide num_kv_heads (if available)
+    - TEP and DEP must divide num_experts (if available)
+    """
+    is_moe = bool(model_info.is_moe)
+    num_kv_heads = model_info.num_kv_heads
+    num_experts = model_info.num_experts
+    intermediate_size = model_info.intermediate_size
+    quant_block = model_info.quantization_block_size
+
+    candidates: list[ParallelizationMapping] = []
+    if is_moe:
+        if phase == "prefill":
+            candidates = [ParallelizationMapping(tep=num_gpus)]
+        else:
+            candidates = [ParallelizationMapping(dep=num_gpus)]
+    else:
+        candidates = [ParallelizationMapping(tp=num_gpus)]
+
+    # now verify if the candidates are valid
+    verified: list[ParallelizationMapping] = []
+    for m in candidates:
+        # 1) KV heads divisibility checks
+        if m.tp is not None:
+            if num_kv_heads is None:
+                logger.warning(
+                    f"Skipping KV heads divisibility check for TP={m.tp}: num_kv_heads is unknown"
+                )
+            else:
+                if int(num_kv_heads) % int(m.tp) != 0:
+                    logger.warning(
+                        f"Invalid mapping TP={m.tp}: num_kv_heads={num_kv_heads} not divisible by TP"
+                    )
+                    continue
+
+        if m.tep is not None:
+            if num_kv_heads is None:
+                logger.warning(
+                    f"Skipping KV heads divisibility check for TEP={m.tep}: num_kv_heads is unknown"
+                )
+            else:
+                if int(num_kv_heads) % int(m.tep) != 0:
+                    logger.warning(
+                        f"Invalid mapping TEP={m.tep}: num_kv_heads={num_kv_heads} not divisible by TEP"
+                    )
+                    continue
+
+        # 2) Experts divisibility checks (for MoE)
+        if m.tep is not None:
+            if num_experts is None:
+                logger.warning(
+                    f"Skipping experts divisibility check for TEP={m.tep}: num_experts is unknown"
+                )
+            else:
+                if int(num_experts) % int(m.tep) != 0:
+                    logger.warning(
+                        f"Invalid mapping TEP={m.tep}: num_experts={num_experts} not divisible by TEP"
+                    )
+                    continue
+
+        if m.dep is not None:
+            if num_experts is None:
+                logger.warning(
+                    f"Skipping experts divisibility check for DEP={m.dep}: num_experts is unknown"
+                )
+            else:
+                if int(num_experts) % int(m.dep) != 0:
+                    logger.warning(
+                        f"Invalid mapping DEP={m.dep}: num_experts={num_experts} not divisible by DEP"
+                    )
+                    continue
```

Comment:
This can be standardized into a helper function to largely reduce the number of lines of code